pkg/policy/selectorcache.go

```diff
@@ -439,6 +440,49 @@ func (f *fqdnSelector) notifyUsers(sc *SelectorCache, added, deleted []identity.
 	}
 }
 
+// allocateIdentityMappings is an initializer for 'fqdnSelector' that takes a
+// slice of IPs that should be associated with the selector and caches those
+// selections. No notifications are propagated to the users.
+//
+// This function may be called at initialization for a new fqdnSelector, or
+// upon new use of an FQDN selector with the same string representation of a
+// previously-cached selector (for example the same toFQDNs match pattern in
+// different rules).
+func (f *fqdnSelector) allocateIdentityMappings(idAllocator cache.IdentityAllocator, selectorIPMapping map[api.FQDNSelector][]net.IP) {
+	var (
+		currentlyAllocatedIdentities []*identity.Identity
+		selectorIPs                  []net.IP
+		ids                          []identity.NumericIdentity
+		err                          error
+	)
+
+	// Allocate identities for each IPNet and then map to selector
+	selectorIPs = selectorIPMapping[f.selector]
+	log.WithFields(logrus.Fields{
+		"fqdnSelector": f.selector,
+		"ips":          selectorIPs,
+	}).Debug("getting identities for IPs associated with FQDNSelector")
+
+	// TODO: Consider if upserts to ipcache should be delayed until endpoint policies have been
+	// updated. This is the path from policy updates rather than for DNS proxy results. Hence
+	// any existing IPs would typically already have been pushed to the ipcache as they would
+	// not be newly allocated. We need the 'allocation' here to get a reference count on the
+	// allocations.
+	if currentlyAllocatedIdentities, err = idAllocator.AllocateCIDRsForIPs(selectorIPs, nil); err != nil {
+		log.WithError(err).WithField("prefixes", selectorIPs).Warn(
+			"failed to allocate identities for IPs")
+		return
+	}
+	ids = make([]identity.NumericIdentity, 0, len(currentlyAllocatedIdentities))
+	for i := range currentlyAllocatedIdentities {
+		ids = append(ids, currentlyAllocatedIdentities[i].ID)
+	}
+
+	for _, id := range ids {
+		f.cachedSelections[id] = struct{}{}
+	}
```

Comment:
Ah I noticed this is just copied code from the name manager. Since this has already existed in the codebase, I'd lean towards lowering the severity of this concern.

Reply:
FWIW, the slice is only `uint32` in the end and keeping in mind that FQDN policies result in `/32` CIDR identities, which is to say that this may not even be much bloat anyway.

Reply:
Good point, yeah I was just refactoring this code but this seems reasonable to clean up.

Reply:
https://github.com/cilium/cilium/pull/17959